Sources/MapboxCoreNavigation/MapMatchingResult.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+import MapboxNavigationNative
+import CoreLocation
+
+/// Provides information about the status of the enhanced location updates generated by the map matching engine of the Navigation SDK.
+public struct MapMatchingResult {
+    
+    /// The best possible location update, snapped to the route or map matched to the road if possible
+    public var enhancedLocation: FixLocation
+    
+    /// A list of predicted location points leading up to the target update.
+    ///
+    /// The last point on the list (if it is not empty) is always equal to `enhancedLocation`.
+    public var keyPoints: [FixLocation]
+    
+    /// Whether the SDK thinks that the user is off road.
+    ///
+    /// Based on the `offRoadProbability`.
+    public var isOffRoad: Bool
+    
+    /// Probability that the user is off road.
+    public var offRoadProbability: Float
+    
+    /// Returns true if map matcher changed its opinion about most probable path on last update.
+    ///
+    /// In practice it means we don't need to animate puck movement from previous to current location and just do an immediate transition instead.
+    public var isTeleport: Bool
+    
+    /// When map matcher snaps to a road, this is the confidence in the chosen edge from all nearest edges.
+    public var roadEdgeMatchProbability: Float
+    
+    init(status: NavigationStatus) {
+        enhancedLocation = status.location
+        keyPoints = status.keyPoints
+        isOffRoad = status.offRoadProba > 0.5
+        offRoadProbability = status.offRoadProba
+        isTeleport = status.mapMatcherOutput.isIsTeleport
+        roadEdgeMatchProbability = status.mapMatcherOutput.matches.first?.proba ?? 0.0
```

Comment:
Map matching result should contain relevant information, so it is focused on providing data for the most recent update. Probabilities for passed locations seems useless at this point. However, it is provided on each status update, so if user still wants to collect history data - he still can store it.

Reply:
I'm not talking about historical data. Just that `status.mapMatcherOutput.matches` have more data than we provide here. But it might be implemented in other PRs as an addition to these fields.

Reply:
`status.mapMatcherOutput.matches.first` is the latest added match, current user position. This array is prepended on each tick, so there is no need to traverse it all, because previous values (by index 1,2,3...) were already reported by previous MapMatchingResults.